src/packet/packetlist.js

```diff
@@ -30,7 +30,7 @@ class PacketList extends Array {
    * Reads a stream of binary data and interprets it as a list of packets.
    * @param {Uint8Array | ReadableStream<Uint8Array>} bytes A Uint8Array of bytes.
    */
-  async read(bytes, allowedPackets, streaming) {
+  async read(bytes, allowedPackets, streaming, config = defaultConfig) {
```

Comment:
In most cases, `streaming` is still preceded by optional parameters, so when passing down the `config` to lower-level functions, those values still need to be specified, hence I don't think the patch would be substantially cleaner/simpler in the end (or what did you mean?).

About the current `packet.read` change where `streaming` is last, I'd really consider that to be temporary, given the corresponding discussion in https://github.com/openpgpjs/openpgpjs/pull/1241#discussion_r578686693 .

I also prefer to keep `config` as last param, first to avoid breaking (potentially most) existing code that uses the public API, but also because I see it as the param that has the most marginal effect.